test/transforms/index.js

```diff
@@ -177,4 +177,25 @@ describe('transforms', async () => {
       })
     }
   })
+
+  describe('set-data', () => {
+    const dir = resolve(__dirname, './fixtures/set-data')
+    const tests = fs.readdirSync(dir)
+
+    for (const test of tests) {
+      if (test[0] == '.') continue
+
+      it(test, async () => {
+        const testDir = resolve(dir, test)
+        const fn = require(testDir).default
+        const input = await readYaml(resolve(testDir, 'input.yaml'))
+        const expected = await readYaml(resolve(testDir, 'output.yaml'))
+
+        let state = Raw.deserialize(input, { terse: true })
+        state = fn(state)
+        const json = Array.from(state.data.entries())
```

Comment:
For these tests, let's stick with the `Raw.serialize(state, { terse: true })` like the other tests for now, so that they don't deviate and get confusing.